test/e2e/fixtures/when.go

```diff
@@ -108,10 +105,17 @@ func (w *When) WaitForWorkflowCondition(test func(wf *wfv1.Workflow) bool, condi
 }
 
 func (w *When) waitForWorkflow(workflowName string, test func(wf *wfv1.Workflow) bool, condition string, timeout time.Duration) *When {
+	w.t.Helper()
 	start := time.Now()
-	logCtx := log.WithFields(log.Fields{"workflow": workflowName, "condition": condition, "timeout": timeout})
+
+	fieldSelector := ""
```

Comment:
This changes the behaviour of this func to make it more useful, you no longer need to know the name of the workflow you want to wait for. We alternatively wait for any `argo-e2e` workflow.